File: Dec-Jan-Feb-Mar/Java 6-7 08-02 to 21-03/12-02/Operators.java
class Operators {
    public static void main(String[] args) {
        // Arithmetic operators
        int a = 10, b = 4;

        System.out.println(a+b); // 14
        System.out.println(a/b); // 2
        System.out.println(a%b); // 2

        // Comparison operators
        System.out.println(a < b); // false
        System.out.println(a >= b); // true
        System.out.println(a == 50); // false
        System.out.println(a != b); // true

        // Logical operators
        System.out.println((a < b) && (b > 5)); // false
        System.out.println((a > b) || (b < a)); // true

        // bitwise operators
        System.out.println(7 & 3); // 111 & 011 ===>  011 (3)
        System.out.println(5 ^ 2); // 101 ^ 010 ===>  111  (7)

        // shift operators
        System.out.println(7 << 1); // 111 << 1  ===> 1110
        System.out.println(10 >> 1); // 1010 >> 1  ===> 101

        // unary operators
        int x = 10;

        System.out.println(++x); // 11
        System.out.println(x);   // 11
        System.out.println(x++); // 11

        System.out.println(x--); // 12
        System.out.println(x);  //  11
        System.out.println(--x); // 10
    }
}

File: FE 7-8 24-11 to/JS/05-01/operators.js
// Operators are used to do operational calculations on given operands.
// syntax: operand operator operand (5 + 3)

// Types of Operators
// Arithmetic operators (returns a value of Number data type)
// (addition +, subtraction -, multiplication *, division /, modulus %, power **)
let a = 5, b = 8;

console.log(`Addition of ${a} and ${b} is ${a+b}`);
console.log(`Multiplication of ${a} and ${b} is ${a*b}`);
console.log(`Division of ${a} and ${b} is ${b/a}`);
console.log(`Modulo of ${a} and ${b} is ${b%a}`);


// Assignment operators (used to assign some value to a variable)
// (=, +=, -=, *=, /=, %=)
let x = 15, y = 21;
// multiply x value with 2 and store the result again in x
// x = x * 2
x *= 2;
console.log(`x value after multiplication is: ${x}`);


// Comparison operators (used to compare two values and return boolean value as a result)
// (==, <, <=, >, >=, !=, ===, !==)
let n1 = 45, n2 = 9, n3 = "9";
// check whether both n1 and n2 value are equal
console.log(n1 == n2); // false
// compare n2 and n3 by value
console.log(n2 == n3); // true
console.log(n2 != n3); // false

// compare n2 and n3 by value and type also
console.log(n2 === n3); // false
console.log(n2 !== n3); // true

console.log(n1 >= n3); // true


// Logical operators (used on boolean operands to return a boolean result)
// (AND - &&), (OR - ||)
let x1 = 3, y1 = 8;
// define whether both the coordinates lies in positive quarter of the graph
let ans = (x1 > 0 && y1 > 0);
console.log(`Both co-ordinates lies in first quarter?: ${ans}`);

let num = 10;
// print whether given number is divisble by 3 or 5
let isDivisible = (num%3 == 0 || num%5 == 0);
console.log(`Number is divisible by either 3 or 5?: ${isDivisible}`);